src/prefect_server/utilities/collections.py

```diff
@@ -19,3 +20,61 @@ def chunked_iterable(iterable: Iterable, size: int):
         if not chunk:
             break
         yield chunk
+
+
+class TimedUniqueValueStore:
+    """
+    Stores unique values for a specific amount of time.
+
+    Args:
+        - duration (float, int): the number of seconds to store a value before removing it
+    """
+
+    def __init__(self, duration: Union[float, int] = 10):
+        self.values = set()
+        self.lock = threading.Lock()
+        self.duration = duration
+
+    def add_and_expire(self, value: Hashable, duration: Optional[float, int] = None) -> bool:
```

Comment:
It doesn't seem like we actually need the `duration` override here for any of the current use-cases. can we just call this `add`?

Reply:
yes can do 👍